src/contentful/revalidate.ts

```diff
@@ -0,0 +1,81 @@
+import type { CMS_ITEM_VALUE, InputObject, OutputObject, WebHookHandler } from '../types'
+
+interface ContentfulWebhookResponse {
+  sys: {
+    /*
+      Entity id
+    */
+    id: string
+    type: string
+    contentType: {
+      sys: {
+        type: 'Link'
+        linkType: 'ContentType'
+        id: string
+      }
+    }
+  }
+  fields: {
+    [key: string]: { 'en-US': string } | { [key: string]: string }
+  }
+}
+
+export const revalidate: WebHookHandler<ContentfulWebhookResponse, unknown> = async (
+  request,
+  cb
+) => {
+  if (process.env?.TELEPORTHQ_ISR_TOKEN !== request.query?.['TELEPORTHQ_ISR_TOKEN']) {
+    return
+  }
+
+  const content = request.body
+  const contentType = content.sys?.contentType?.sys?.id
+
+  switch (content.sys.type) {
+    case 'DeletedEntry': {
+      /*
+        For deletion of a contentful entry, we have access to only the `id`
+        If the URL's are using any other fields in them. We can't clear those paths.
+      */
+      cb({ id: content.sys.id }, contentType)
+      break
+    }
+    case 'Entry': {
+      const normalizedData: Record<string, CMS_ITEM_VALUE | Record<string, CMS_ITEM_VALUE>> = {
+        id: content.sys.id,
+        ...transformInput(content.fields),
+      }
+
+      cb(normalizedData, contentType)
+      break
+    }
+    default: {
+      throw new Error(
+        `[ON-DEMAND_ISR]: Recevied a webhook operation that is not supported ${contentType}`
+      )
+    }
+  }
+}
+
+const transformInput = (input: InputObject): OutputObject => {
```

Comment:
isn't this the same as the normalize function used when fetching entities for CMS Lists and Items? Is there any overlap between the two? If yes, we should use the same function call to map the data, instead of duplicating the mapping logic into two places.

Reply:
Nope, it is sending in a complete different way. In usual responses they are not sending `language` as keys. But in webhooks they send it differently. Tried the existing ones. Didn't work.

Reply:
but now we'll have to maintain two different mappings. Did you sync with @elizamaria on this?

Reply:
Yes i did, indeed i discussed with @ionutpasca too on this language identifier.